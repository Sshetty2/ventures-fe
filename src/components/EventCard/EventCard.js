import React from 'react';
import { Link } from 'react-router-dom';

const EventCard = ({ id, name, image, city, state, date, pathname }) => {
  return (
    <Link to={pathname === '/' ? `/event/${id}` : `/profile/event/${id}`}>
      <div className='EventCard' style={{ backgroundImage: `url(${image})`}}>
        <div className='text-container'>
          <h3 className='card-name'>{name}</h3>
          <h5 className='card-location'>{city}, {state}</h5>
          <h5 className='card-date'>{date}</h5>
        </div>
      </div>
    </Link>
  )
}

export default EventCard;