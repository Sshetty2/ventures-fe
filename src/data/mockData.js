export const mockEvents = {
  data: [
    {
      id: "1",
      type: "event",
      attributes: {
        name: "Shelf Road Craggin' Classic",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "2",
      type: "event",
      attributes: {
        name: "Hueco Rock Rodeo",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "3",
      type: "event",
      attributes: {
        name: "Flash Foxy Women's Climbing Festival",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "4",
      type: "event",
      attributes: {
        name: "Flash Foxy Summerfest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "5",
      type: "event",
      attributes: {
        name: "International Climber's Festival",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "6",
      type: "event",
      attributes: {
        name: "Leavenworth Rock Fest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "3",
      type: "event",
      attributes: {
        name: "Joes Valley Fest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    }
  ]
};

export const mockUser = {
  name: 'Jane Doe',
  image_url: 'https://alumni.turing.io/sites/default/files/styles/graduate_full_profile/public/squareheadshot1.jpg?itok=EGlR-8YR',
  bio: 'Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!!'
}

export const mockUserEvents = [
  {
    id: "6",
    type: "event",
    attributes: {
      name: "Leavenworth Rock Fest",
      city: "Canon Springs",
      state: "CO",
      event_type: "climbing",
      price: 35,
      start_date: "10-19-2018",
      end_date: "10-21-2018",
      description:
        "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
      event_url: "https://americanalpineclub.org/shelf-road-cc",
      image_url:
        "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
      video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo",
      status: 'attending'
    }
  },
  {
    id: "3",
    type: "event",
    attributes: {
      name: "Joes Valley Fest",
      city: "Canon Springs",
      state: "CO",
      event_type: "climbing",
      price: 35,
      start_date: "10-19-2018",
      end_date: "10-21-2018",
      description:
        "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
      event_url: "https://americanalpineclub.org/shelf-road-cc",
      image_url:
        "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
      video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo",
      status: 'wishlist'
    }
  }
]