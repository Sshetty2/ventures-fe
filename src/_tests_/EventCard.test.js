import React from 'react';
import { shallow } from 'enzyme';
import EventCard from '../components/EventCard/EventCard';
import { mockEvents } from '../data/mockData';

const mockProps = {
  id: mockEvents.data[0].id,
  key: mockEvents.data[0].id,
  name: mockEvents.data[0].attributes.name,
  image: mockEvents.data[0].attributes.image_url,
  city: mockEvents.data[0].attributes.city,
  state: mockEvents.data[0].attributes.state,
  date: mockEvents.data[0].attributes.start_date,
  pathname: '/'
}

describe('EventCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<EventCard {...mockProps} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});