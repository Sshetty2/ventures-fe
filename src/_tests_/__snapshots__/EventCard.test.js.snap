// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should match the snapshot 1`] = `
<Link
  to="/event/1"
>
  <div
    className="EventCard"
    style={
      Object {
        "backgroundImage": "url(https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg)",
      }
    }
  >
    <div
      className="text-container"
    >
      <h3
        className="card-name"
      >
        Shelf Road Craggin' Classic
      </h3>
      <h5
        className="card-location"
      >
        Canon Springs
        , 
        CO
      </h5>
      <h5
        className="card-date"
      >
        10-19-2018
      </h5>
    </div>
  </div>
</Link>
`;
